File: Java/src/ComputeMultipleSum.java
import java.util.stream.IntStream;

public class ComputeMultipleSum {
    public static void main(String[] args) {
        System.out.println(computeMultipleSum(1000)); // Example usage
    }

    public static int computeMultipleSum(int n) {
        if (n <= 0 || n > 1000) {
            throw new IllegalArgumentException("n must be greater than 0 and less than or equal to 1000");
        }

        return IntStream.rangeClosed(1, n)
                .filter(i -> i % 3 == 0 || i % 5 == 0 || i % 7 == 0)
                .sum();
    }
}

File: Java/kotlin/ComputeMultipleSum.kt
package kotlin

object ComputeMultipleSum {
    @JvmStatic
    fun main(args: Array<String>) {
        println(computeMultipleSum(-11)) // Example usage
    }

    fun computeMultipleSum(n: Int): Int {
        require(n > 0 && n <= 1000) {
            "n must be greater than 0 and less than or equal to 1000"
        }

        return (1..n)
            .filter { it % 3 == 0 || it % 5 == 0 || it % 7 == 0 }
            .sum()
    }
}

